admin/aws.py

```diff
@@ -281,11 +284,52 @@ def perform_download_s3_key(dispatcher, intent):
 
     bucket = s3.get_bucket(intent.source_bucket)
     key = bucket.get_key(intent.source_key)
+    if key is None:
+        raise KeyError(intent.source_key)
     with intent.target_path.open('w') as target_file:
         key.get_contents_to_file(target_file)
 
 
 @attributes([
+    "source_bucket",
+    "source_key",
+])
+class ReadS3Key(object):
+    """
+    Read a file from S3.
+
+    :ivar bytes source_bucket: Name of bucket to read key from.
+    :ivar bytes source_key: Name of key to read.
+    """
+
+
+@sync_performer
+@do
+def perform_read_s3_key(dispatcher, intent):
+    """
+    See :class:`ReadS3Key`.
+    """
+    target_file = FilePath(
+        u'/tmp/{}.perform_read_s3_key'.format(
+            __file__.replace(u"/", "!"),
+        )
+    ).temporarySibling()
```

Comment:
Looking at `DownloadS3Key`, it would be less code and simpler to just download to a string (which `boto` can do), than do some gyrations with a temporary file, just to reuse a bit of code.